README.md

```diff
@@ -28,7 +28,34 @@ It knows to do this because it sees a top level function containing `Debug.todo
 This rule isn't limited to generating toString functions for custom types though.
 
 Here are some other type signatures it looks for
-* `String -> Maybe MyType` generates a fromString function
-* `Codec e MyType` generates a MartinSStewart/elm-serialize Codec
-* `List MyType` generates a list containing all the custom type variants
-* `Random.Generator MyType` generates an elm/random generator (unfinished)
\ No newline at end of file
+
+- `String -> Maybe MyType` generates a fromString function
+- `Codec e MyType` generates a MartinSStewart/elm-serialize Codec
+- `List MyType` generates a list containing all the custom type variants
+- `Random.Generator MyType` generates an elm/random generator (unfinished)
+
+## Notes
+
+To be transformed into actual documentation...
+
+### Function resolution priority
+
+One way to think about the code generation the rule here does is a systematic way of composing other functions into larger units.
+In order to do that, it has to get a good idea of what those functions are.
+
+The way we solve this problem is to search your codebase for functions that have the right kind of signature, then use them in broader compositions.
```

Comment:
Maybe if the function signature is too "simple" (this could be defined as a function signature that only contain core types, i.e. `String -> Maybe Int`) then no function gets used due to the risk of using the wrong thing being too great?

That said, I'm not sure if this applies to any of the existing rules. They all have some custom type in them such as Codec?

Eitherway, this isn't a blocker for this PR.

Reply:
I was actually thinking that perhaps the feature of automatically picking up definitions (I think that whether it's from the users own code or from dependencies actually doesn't matter) might be something that the CodeGenerator author might want to disable? We could probably make a flag for that... But perhaps it would be good to see if the issue is a problem in practice first?